File: jfingerprintmanager/src/main/java/com/jesusm/jfingerprintmanager/encryption/EncryptionData.java
package com.jesusm.jfingerprintmanager.encryption;

import com.jesusm.jfingerprintmanager.utils.TextUtils;

class EncryptionData {
    private static final String SEPARATOR = ":";

    private String encryptedMessage, encryptedIVs;
    private Encoder encoder;

    EncryptionData(byte[] message, byte[] IVs, Encoder encoder) {
        this.encoder = encoder;
        this.encryptedMessage = encoder.encode(message);
        this.encryptedIVs = encoder.encode(IVs);
    }

    EncryptionData(String encryptedMessage, Encoder encoder) {
        this.encoder = encoder;
        String[] parameters = encryptedMessage.split(SEPARATOR);
        this.encryptedMessage = parameters[0];
        if (parameters.length > 1) {
            this.encryptedIVs = parameters[1];
        }
    }

    byte[] message()
    {
        return encoder.decode(encryptedMessage);
    }

    byte[] getIVs() {
        return encoder.decode(encryptedIVs);
    }

    String printEncryptedInformation() {
        return encryptedMessage + SEPARATOR + encryptedIVs;
    }

    boolean dataIsCorrect() {
        TextUtils textUtils = new TextUtils();
        return !textUtils.isEmpty(encryptedMessage) && !textUtils.isEmpty(encryptedIVs);
    }
}

File: kfingerprintmanager/src/main/kotlin/com/jesusm/kfingerprintmanager/encryption/EncryptionData.kt
package com.jesusm.kfingerprintmanager.encryption

class EncryptionData(var encryptedMessage: String, var encryptedIVs: String?,
                     val encoder: Encoder, val separator: String = ":") {

    constructor(encryptedMessageString: String, encoder: Encoder) :
            this(encryptedMessageString, encryptedMessageString, encoder, ":") {
        val split = encryptedMessageString.split(separator)
        encryptedMessage = split[0]
        if (split.size > 1) {
            encryptedIVs = split[1]
        } else {
            encryptedIVs = null
        }
    }

    constructor(encryptedMessage: ByteArray, encryptedIVs: ByteArray, encoder: Encoder) :
            this(encoder.encode(encryptedMessage), encoder.encode(encryptedIVs), encoder)

    fun print(): String = encryptedMessage + separator + encryptedIVs
    fun dataIsCorrect(): Boolean = separator.isNullOrEmpty().not() && encryptedIVs.isNullOrEmpty().not()

    fun decodedMessage(): ByteArray = encoder.decode(encryptedMessage)
    fun decodedIVs(): ByteArray? = if (encryptedIVs != null) encoder.decode(encryptedIVs as String) else null
}